src/Controllers/HomeController.cs

```diff
@@ -1,21 +1,35 @@
 ﻿using System.Diagnostics;
+using System.Collections.Generic;
 using GovUk.Education.SearchAndCompare.UI.ViewModels;
 using GovUk.Education.SearchAndCompare.UI.Services;
 using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Mvc;
 using GovUk.Education.SearchAndCompare.UI.ActionFilters;
+using Microsoft.Extensions.Configuration;
 
 namespace GovUk.Education.SearchAndCompare.UI.Controllers
 {
     //[Authorize]
     public class HomeController : CommonAttributesControllerBase
     {
+        private bool PreLaunchMode;
+
+        public HomeController(IConfiguration Configuration)
+        {
+            PreLaunchMode = !string.IsNullOrWhiteSpace(Configuration["SITE_PASSWORD"]);
```

Comment:
code duplication

reject

Reply:
@timabell I paired with @the-daniel-rothig and we couldn't figure out a different way to do this. We need a way to pass this down from `Startup.cs`, or a different way to switch the route around.

Reply:
Also, while it is duplication, may I invoke the [rule of three](http://wiki.c2.com/?ThreeStrikesAndYouRefactor)? 😬